File: examples/hello-world/gabriel-greeting-ruby/api/public_test.rb
# frozen_string_literal: true

require "minitest/autorun"
require_relative "../support"
require "v1/greeting_pb"
require_relative "public"

class GreetingPublicTest < Minitest::Test
  def test_list_languages_includes_english
    response = GabrielGreetingRuby::Api::Public.list_languages(
      Greeting::V1::ListLanguagesRequest.new
    )

    assert_equal 56, response.languages.length
    english = response.languages.find { |language| language.code == "en" }
    refute_nil english
    assert_equal "English", english.name
    assert_equal "English", english.native
  end

  def test_say_hello_uses_requested_language
    response = GabrielGreetingRuby::Api::Public.say_hello(
      Greeting::V1::SayHelloRequest.new(name: "Bob", lang_code: "fr")
    )

    assert_equal "Bonjour Bob", response.greeting
    assert_equal "French", response.language
    assert_equal "fr", response.lang_code
  end

  def test_say_hello_uses_localized_default_name
    response = GabrielGreetingRuby::Api::Public.say_hello(
      Greeting::V1::SayHelloRequest.new(lang_code: "ja")
    )

    assert_equal "こんにちは、マリアさん", response.greeting
    assert_equal "Japanese", response.language
    assert_equal "ja", response.lang_code
  end

  def test_say_hello_falls_back_to_english
    response = GabrielGreetingRuby::Api::Public.say_hello(
      Greeting::V1::SayHelloRequest.new(lang_code: "unknown")
    )

    assert_equal "Hello Mary", response.greeting
    assert_equal "English", response.language
    assert_equal "en", response.lang_code
  end
end

File: examples/hello-world/gabriel-greeting-python/api/public_test.py
from __future__ import annotations

import unittest

from support import ensure_import_paths

ensure_import_paths()

from v1 import greeting_pb2

from api import public


class GreetingPublicTest(unittest.TestCase):
    def test_list_languages_includes_english(self) -> None:
        response = public.list_languages(greeting_pb2.ListLanguagesRequest())

        self.assertTrue(response.languages)
        english = next(language for language in response.languages if language.code == "en")
        self.assertEqual(english.name, "English")
        self.assertEqual(english.native, "English")

    def test_say_hello_uses_requested_language(self) -> None:
        response = public.say_hello(
            greeting_pb2.SayHelloRequest(name="Bob", lang_code="fr")
        )

        self.assertEqual(response.greeting, "Bonjour Bob")
        self.assertEqual(response.language, "French")
        self.assertEqual(response.lang_code, "fr")

    def test_say_hello_uses_localized_default_name(self) -> None:
        response = public.say_hello(greeting_pb2.SayHelloRequest(lang_code="ja"))

        self.assertEqual(response.greeting, "こんにちは、マリアさん")
        self.assertEqual(response.language, "Japanese")
        self.assertEqual(response.lang_code, "ja")

    def test_say_hello_falls_back_to_english(self) -> None:
        response = public.say_hello(greeting_pb2.SayHelloRequest(lang_code="unknown"))

        self.assertEqual(response.greeting, "Hello Mary")
        self.assertEqual(response.language, "English")
        self.assertEqual(response.lang_code, "en")


if __name__ == "__main__":
    unittest.main()
